public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/AddAnomalyDetector.tsx

```diff
@@ -0,0 +1,740 @@
+    field.onChange(e);
+    setDetectorNameFromVis(e.target.value);
+  };
+  const onIntervalChange = (e, field) => {
+    field.onChange(e);
+    setIntervalalue(e.target.value);
+  };
+  const onDelayChange = (e, field) => {
+    field.onChange(e);
+    setDelayValue(e.target.value);
+  };
+  const aggList = embeddable.vis.data.aggs.aggs.filter(
+    (feature) => feature.schema == 'metric'
+  );
+  const featureList = aggList.filter(
+    (feature, index) =>
+      index <
+      (aggList.length < MAX_FEATURE_NUM ? aggList.length : MAX_FEATURE_NUM)
+  );
+
+  const getFeatureNameFromVisParams = (id) => {
+    let name = find(embeddable.vis.params.seriesParams, function (param) {
+      if (param.data.id === id) {
+        return true;
+      }
+    });
+
+    return name.data.label.replace(/[^a-zA-Z0-9-_]/g, '_');
+  };
+
+  const notifications = getNotifications();
+
+  const handleValidationAndSubmit = (formikProps) => {
+    if (!isEmpty(formikProps.errors)) {
+      focusOnFirstWrongFeature(formikProps.errors, formikProps.setFieldTouched);
+      notifications.toasts.addDanger('One or more input fields is invalid');
+    } else {
+      handleSubmit(formikProps);
+    }
+  };
+
+  const handleSubmit = (formikProps) => {
+    formikProps.setSubmitting(true);
+    try {
+      const detectorToCreate = formikToDetector(formikProps.values);
+      dispatch(createDetector(detectorToCreate))
+        .then(async (response) => {
+          notifications.toasts.addSuccess(
+            `Detector created: ${formikProps.values.name}`
+          );
+          dispatch(startDetector(response.response.id))
+            .then((startDetectorResponse) => {
+              notifications.toasts.addSuccess(
+                `Successfully started the real-time detector`
+              );
+            })
+            .catch((err: any) => {
+              notifications.toasts.addDanger(
+                prettifyErrorMessage(
+                  getErrorMessage(
+                    err,
+                    'There was a problem starting the real-time detector'
+                  )
+                )
+              );
+            });
+          enum VisLayerTypes {
+            PointInTimeEvents = 'PointInTimeEvents',
+          }
+          const fn = {
+            type: VisLayerTypes.PointInTimeEvents,
+            name: 'overlay_anomalies',
+            args: {
+              detectorId: response.response.id,
+            },
+          } as VisLayerExpressionFn;
+
+          const savedObjectToCreate = {
+            title: embeddable.vis.title,
+            pluginResourceId: response.response.id,
+            visId: embeddable.vis.id,
+            savedObjectType: 'visualization',
+            visLayerExpressionFn: fn,
+          } as ISavedAugmentVis;
+
+          const savedObject = await createAugmentVisSavedObject(
+            savedObjectToCreate
+          );
+
+          const saveObjectResponse = await savedObject.save({});
```

Comment:
We still will need to have error handling of this part. Refer to lines 287-311 of [this file](public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/AddAnomalyDetector.tsx). 

@amitgalitz @jackiehanyang can you work together to have this be consistent across both create and associate flows?

Reply:
Yes do you think it should be a part of this PR or I can add this in a separate PR when I add the alerting integration on the notification. For here Jackie could copy what I have done in the associate existing, the link you sent is broken for me but I assume its pointing to my PR in the file AddAnomalyDetector. Jackie could add that and then when I rebase with her changes I can refactor it to a helper fn and make sure they are the same. I can then also add more logic with the set up alerting notification

Reply:
That sounds like a plan to me :), yes you're correct in the file I was referring to. For some reason those github links have been broken as of late...

Reply:
Can you either copy the lines I have or add a TODO in code with what I should (better do document so nothing is forgotten)